Libraries/LibGC/RootVector.h

```diff
@@ -46,7 +46,7 @@ class RootVector final
     {
     }
 
-    virtual ~RootVector() = default;
+    ~RootVector() = default;
```

Comment:
Presumably the virtual is here because the author thought the base class had a virtual destructor and they just forgot override? 

I feel like actually conforming these guys to our style guide would have caught the odd behavior here.

Reply:
nothing that's actionable here, though, right?

Reply:
I... Don't know. Is the RootVectorBase dtor actually called? I don't actually know the behavior of non-virtual base destructors. It's very likely that no one will ever try to delete these guys through a pointer to base though, so it's probably fine

Reply:
> It's very likely that no one will ever try to delete these guys through a pointer to base though

You can't, as long as the destructor is `protected`.

If you could and would, calling the destructor when the static type is the non-virtual base (e.g. `~RootVectorBase()`) would not destruct the derived class's members.